### ISSUE TITLE
Disable Spreads Button

### ISSUE THREAD

**Issue by imskrishnan** [stephanrauh/ngx-extended-pdf-viewer#2645]
Hi @stephanrauh, wanted to know if there are any APIs to make the spreads button disabled by any chance? I see only 3 states to set the spread mode - off, even or odd. But I want to disable those buttons, as in grey out the colors and have no pointer-events.

**Comment by imskrishnan** [comment]
I believe this is something that was available in pdf version 15, where if I selected horizontal scrolling, all the spreads button would get disabled - greyed out and non-clickable. Is this changed in pdf version 21?

**Comment by Julian-B90** [comment]
Hey @imskrishnan 

Her you can see when `scrollMode` is `1` the Spread Button is disabled.

https://github.com/stephanrauh/ngx-extended-pdf-viewer/blob/15706db7dff42ade61db94a58312decddc3b6466/projects/ngx-extended-pdf-viewer/src/lib/toolbar/pdf-odd-spread/pdf-odd-spread.component.html#L11C3-L11C32

**Comment by imskrishnan** [comment]
Hi @Julian-B90,

I see this [disabled] attribute only for the <pdf-shy-button> component and not for the <ngx-extended-pdf-viewer> component.

```
<ngx-extended-pdf-viewer
              *ngIf="pdfSrc"
              (textLayerRendered)="onTextLayerRendered($event)"
              (updateFindMatchesCount)="pdfSearchService.updateFindResultMatchesCount($event)"
              (pdfLoaded)="onPdfLoadEvent()"
              [src]="pdfSrc"
              [useBrowserLocale]="true"
              [showRotateButton]="true"
              [showOpenFileButton]="false"
              [showHandToolButton]="false"
              [showPrintButton]="false"
              [showDownloadButton]="false"
              [showFindButton]="false"
              [handTool]="false"
              [sidebarVisible]="false"
              [showSidebarButton]="false"
              [showSecondaryToolbarButton]="true"
              [showDrawEditor]="false"
              [showStampEditor]="false"
              [showTextEditor]="false"
              [showHighlightEditor] ="false"
              id="ngx-extended-pdf-viewer"
              class="view-container"
              theme="dark"
              (pageChange)="pageNavigationService.updateCurrentPage($event)"
              [backgroundColor]="pdfViewerBackgroundColor"
              [enableDragAndDrop]="false"
      >
      </ngx-extended-pdf-viewer>
```

is it possible to specify here?

**Comment by Julian-B90** [comment]
This is the input https://github.com/stephanrauh/ngx-extended-pdf-viewer/blob/15706db7dff42ade61db94a58312decddc3b6466/projects/ngx-extended-pdf-viewer/src/lib/ngx-extended-pdf-viewer.component.ts#L307

you can add  `[scrollMode]="1"`  to ngx-extended-pdf-viewer

Or you can use the Enum ScrollModeType: 

``` 
export enum ScrollModeType {
  vertical = 0,
  horizontal = 1,
  wrapped = 2,
  page = 3,
}
```

**Comment by imskrishnan** [comment]
Hi @Julian-B90,

So, this only sets the scroll mode to horizontal when the app loads, but doesn't disable the spread buttons. I have tried setting [disabled]="true" as well but its not disabling the spread buttons. This is how the button looks like in the dom. Its more like setting the `disabled` attribute to this button.

```
<button _ngcontent-ng-c3098786904="" type="button" class="secondaryToolbarButton toggled always-in-secondary-menu ng-star-inserted" id="secondarySpreadNone" ng-reflect-ng-class="always-in-secondary-menu" title="Do not join page spreads" data-l10n-id="pdfjs-spread-none-button"><span _ngcontent-ng-c3098786904="" role="img" aria-hidden="true" class="icon ng-star-inserted" aria-label="Do not join page spreads"><svg class="pdf-margin-top-3px" width="24px" height="24px" viewBox="0 0 24 24"><path fill="currentColor" d="M6 3c-1 0-1.5.5-1.5 1.5v7c0 1 .5 1.5 1.5 1.5h4c1 0 1.5-.5 1.5-1.5v-7c0-1-.5-1.5-1.5-1.5z"></path></svg></span><!--bindings={
  "ng-reflect-ng-if": "SafeValue must use [property]="
}--><span _ngcontent-ng-c3098786904="" class="toolbar-caption" data-l10n-id="pdfjs-spread-none-button-label">No Spreads</span></button>
```

**Comment by imskrishnan** [comment]
This is what I want to be more precise, the spreads buttons are disabled mode, only when clicking on the horizontal scrolling

![Screenshot 2024-11-11 at 9 12 29 AM](https://github.com/user-attachments/assets/a138e379-13df-4848-b510-81dd6ef54085)

**Comment by stephanrauh** [comment]
That was a fruitful discussion helping me to understand the issue. As it turns out, both of you are right. The button **is** disabled - but only if it's a button, not if it's hiding in the secondary menu. Let's fix that!

<img width="560" alt="image" src="https://github.com/user-attachments/assets/7fafa800-976b-42bd-b3bb-d34697c73c9d">

<img width="538" alt="image" src="https://github.com/user-attachments/assets/df57e7a9-c1bc-4455-a3bc-00c3336a60d5">

**Comment by stephanrauh** [comment]
Your bugfix is going to land with version 22.0.0-alpha.9 soon.  But I won't publish it today - the version needs more testing. 

Stay tuned!
Stephan

**Comment by stephanrauh** [comment]
Your feature has landed with version 22.0.0-alpha.9 a few minutes ago.
